Source/WebCore/rendering/GridMasonryLayout.cpp

```diff
@@ -194,18 +195,78 @@ void GridMasonryLayout::updateItemOffset(const RenderBox& gridItem, LayoutUnit o
 GridArea GridMasonryLayout::gridAreaForIndefiniteGridAxisItem(const RenderBox& item)
 {
     auto itemSpanLength = Style::GridPositionsResolver::spanSizeForAutoPlacedItem(item, gridAxisDirection());
-    auto smallestMaxPos = LayoutUnit::max();
-    unsigned smallestMaxPosLine = 0;
     auto gridAxisLines = m_gridAxisTracksCount + 1;
-    for (unsigned startingLine = 0; startingLine < gridAxisLines - itemSpanLength; startingLine++) {
+
+    // Get item-tolerance from the masonry container's style
+    const auto& tolerance = m_renderGrid->style().itemTolerance();
+
+    if (tolerance.isInfinite()) {
+        // Infinite tolerance: place items strictly in order without considering track lengths
+        // Use round-robin placement starting from the cursor position
+        auto startingLine = m_autoFlowNextCursor;
+
+        // If the item doesn't fit at the cursor position, wrap to the beginning
+        if (startingLine + itemSpanLength > m_gridAxisTracksCount)
+            startingLine = 0;
+
+        auto gridAxisPosition = GridSpan::translatedDefiniteGridSpan(startingLine, startingLine + itemSpanLength);
+        return masonryGridAreaFromGridAxisSpan(gridAxisPosition);
+    }
+
+    // For normal and length-percentage tolerances, find positions within tolerance of the shortest track
+    // Calculate tolerance value
+    auto referenceLength = gridAxisDirection() == Style::GridTrackSizingDirection::Columns
+        ? m_renderGrid->contentBoxLogicalHeight()
+        : m_renderGrid->contentBoxLogicalWidth();
+
+    LayoutUnit toleranceValue = tolerance.switchOn(
+        [&](const CSS::Keyword::Normal&) -> LayoutUnit {
```

Comment:
Since the lambda has `LayoutUnit` as the trailing return value maybe we can can then `return LayoutUnit { ... }` to `return { ....}`?